mosqagent.h

```diff
@@ -56,19 +59,32 @@ struct mosqagent_result {
 
 };
 
+/**
+ * \brief The agent's configuration settings
+ */
 struct mosqagent_config {
-    const char* client_name;
-    const char* host;
+    char* client_name;
+    char* host;
     int port;
 };
 
+/**
+ * If there is already a configuration in the agent, make sure to take care
+ * of the memory object before calling this function, otherwise there may be
+ * memory leaks!
+ */
+int mqtta_load_configuration(struct mosqagent *agent,
```

Comment:
If you use Doxygen style comments here, I would assume an API description. Something like:

> If *agent already had a configuration loaded, the library will close and free it and load the new configuration instead.